src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -0,0 +1,324 @@
+                    using (var stream = new MemoryStream(byteArray))
+                    {
+                        return ReadFromProtobufStream(stream, ServerIssue.Parser).ToArray();
+                    }
+                });
+        }
+
+        public Task<Result<OrganizationResponse[]>> GetOrganizationsAsync(OrganizationRequest request, CancellationToken token)
+        {
+            // Since 6.2; internal
+            var apiPath = BuildRelativeUrl("api/organizations/search",
+                new Dictionary<string, string>
+                {
+                    ["p"] = request.Page.ToString(),
+                    ["ps"] = request.PageSize.ToString()
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => JObject.Parse(stringResponse)["organizations"].ToObject<OrganizationResponse[]>());
+        }
+
+        public Task<Result<PluginResponse[]>> GetPluginsAsync(CancellationToken token)
+        {
+            // Since 2.10; internal
+            var serverPluginsInstalledApiUrl = BuildRelativeUrl("api/updatecenter/installed_plugins");
+
+            return InvokeSonarQubeApi(serverPluginsInstalledApiUrl, token,
+                stringResponse => ProcessJsonResponse<PluginResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<ProjectResponse[]>> GetProjectsAsync(CancellationToken token)
+        {
+            // Since 2.10
+            var projectsApiUrl = BuildRelativeUrl("api/projects/index");
+
+            return InvokeSonarQubeApi(projectsApiUrl, token,
+                stringResponse => ProcessJsonResponse<ProjectResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<PropertyResponse[]>> GetPropertiesAsync(CancellationToken token)
+        {
+            // Since 2.6
+            var propertiesApiUrl = BuildRelativeUrl("api/properties/");
+
+            return InvokeSonarQubeApi(propertiesApiUrl, token,
+                stringResponse => ProcessJsonResponse<PropertyResponse[]>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileChangeLogResponse>> GetQualityProfileChangeLogAsync(
+            QualityProfileChangeLogRequest request, CancellationToken token)
+        {
+            // Since 5.2
+            var apiPath = BuildRelativeUrl("api/qualityprofiles/changelog",
+                new Dictionary<string, string>
+                {
+                    ["profileKey"] = request.QualityProfileKey,
+                    ["ps"] = request.PageSize.ToString()
+                });
+
+            return InvokeSonarQubeApi(apiPath, token,
+                stringResponse => ProcessJsonResponse<QualityProfileChangeLogResponse>(stringResponse, token));
+        }
+
+        public Task<Result<QualityProfileResponse[]>> GetQualityProfilesAsync(QualityProfileRequest request,
+            CancellationToken token)
+        {
+            var queryParams = new Dictionary<string, string>();
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)